Sources/Sentry/SentrySpanId.m

```diff
@@ -77,6 +77,11 @@ + (SentrySpanId *)empty
     return _empty;
 }
 
+- (id)copyWithZone:(nullable NSZone *)zone
+{
+    return [[SentrySpanId alloc] initWithValue:self.value];
```

Comment:
https://developer.apple.com/documentation/objectivec/nsobject/1571945-allocwithzone

```objc
+ (instancetype)allocWithZone:(struct _NSZone *)zone;
```
This method exists for historical reasons; memory zones are no longer used by Objective-C.

And https://developer.apple.com/documentation/foundation/nscopying?language=objc says:

- Implement NSCopying using alloc and init... in classes that don’t inherit copyWithZone:.

It could be.
```suggestion
    return [[[self class] alloc] initWithValue:self.value];
```
Personally, I don't like it.

Reply:
Thanks for the detailed explanation.